python/tests/illarg_tests/numeric_type_fn.py

```diff
@@ -0,0 +1,661 @@
+                      WHERE id = 1"""
+
+
+class illarg_exp_cast_legal(TstView):
+    def __init__(self):
+        # Validated on Postgres
+        self.data = [{"arr": Decimal("0.8693582353988059")}]
+        self.sql = """CREATE MATERIALIZED VIEW exp_cast_legal AS SELECT
+                      EXP(ARR[1]) AS arr
+                      FROM illegal_tbl
+                      WHERE id = 1"""
+
+
+# Negative Test
+class illarg_exp_illegal(TstView):
+    def __init__(self):
+        # checked manually
+        self.sql = """CREATE MATERIALIZED VIEW exp_illegal AS SELECT
+                      EXP(bin) AS bin
+                      FROM illegal_tbl"""
+        self.expected_error = "Cannot apply 'EXP' to arguments of type"
+
+
+# Floor function
+class illarg_floor_legal(TstView):
+    def __init__(self):
+        # Validated on Postgres
+        self.data = [{"reall": Decimal("-1.0"), "decimall": -1}]
+        self.sql = """CREATE MATERIALIZED VIEW floor_legal AS SELECT
+                      FLOOR(reall) AS reall,
+                      FLOOR(decimall) AS decimall
+                      FROM illegal_tbl
+                      WHERE id = 1"""
+
+
+class illarg_floor_cast_legal(TstView):
+    def __init__(self):
+        # Validated on Postgres
+        self.data = [{"arr": -1}]
+        self.sql = """CREATE MATERIALIZED VIEW floor_cast_legal AS SELECT
+                      FLOOR(ARR[1]) AS arr
+                      FROM illegal_tbl
+                      WHERE id = 1"""
+
+
+# Negative Test
+class illarg_floor_illegal(TstView):
+    def __init__(self):
+        # checked manually
+        self.sql = """CREATE MATERIALIZED VIEW floor_illegal AS SELECT
+                      FLOOR(tmestmp) AS tmestmp
+                      FROM illegal_tbl"""
+        self.expected_error = "Cannot apply 'ABS' to arguments of type"
+
+
+# IS_INF function
+class illarg_isinf_legal(TstView):
+    def __init__(self):
+        # checked manually
+        self.data = [{"reall": False, "dbl": None, "decimall": True}]
+        self.sql = """CREATE MATERIALIZED VIEW isinf_legal AS SELECT
+                      IS_INF(reall) AS reall,
+                      IS_INF(dbl) AS dbl,
+                      IS_INF(decimall) AS decimall
+                      FROM illegal_tbl
+                      WHERE id = 2"""
+
+
+class illarg_isinf_cast_legal(TstView):
+    def __init__(self):
+        # checked manually
+        self.data = [{"arr": False}]
+        self.sql = """CREATE MATERIALIZED VIEW isinf_cast_legal AS SELECT
+                      IS_INF(ARR[1]) AS arr
+                      FROM illegal_tbl
+                      WHERE id = 1"""
+
+
+# Negative Test
+class illarg_isinf_illegal(TstView):
+    def __init__(self):
+        # checked manually
+        self.sql = """CREATE MATERIALIZED VIEW isinf_illegal AS SELECT
+                      IS_INF(booll) AS booll
+                      FROM illegal_tbl"""
+        self.expected_error = "Cannot apply 'ABS' to arguments of type"
```

Comment:
something is wrong.
this test should fail, since the function is not ABS.
The message does not match.
Can you see why it does not fail?